diff -ur pgplot/drivers/xtk/tkpgplot.c pgplot.tcl86/drivers/xtk/tkpgplot.c
--- pgplot/drivers/xtk/tkpgplot.c	2014-09-01 19:54:53.597831875 +0200
+++ pgplot.tcl86/drivers/xtk/tkpgplot.c	2014-09-01 19:50:32.158743239 +0200
@@ -2139,7 +2139,8 @@
     tkpg->border = bd;
     tkpg_draw_3d_border(tkpg);
   } else {
-    fprintf(stderr, "Tk_Get3DBorder failed: %s\n", tkpg->interp->result);
+    /*fprintf(stderr, "Tk_Get3DBorder failed: %s\n", tkpg->interp->result);*/
+    fprintf(stderr, "%s\n", "Tk_Get3DBorder failed");
   };
 }
 
