diff -ur pgplot/drivers/xwdriv.c pgplot.new/drivers/xwdriv.c
--- pgplot/drivers/xwdriv.c	2000-12-13 02:30:18.000000000 +0100
+++ pgplot.new/drivers/xwdriv.c	2008-11-13 13:00:32.000000000 +0100
@@ -3344,6 +3344,7 @@
     };
 #else
     if((exe=find_exe(getenv("PGPLOT_DIR"), PGXWIN_SERVER))==NULL &&
+       (exe=find_exe(getenv("/usr/libexec/pgplot"), PGXWIN_SERVER))==NULL &&
        (exe=find_exe(getenv("PATH"), PGXWIN_SERVER))==NULL) {
       fprintf(stderr,
 	  "%s: Couldn't find program \"%s\" in the directory named\n",
diff -ur pgplot/src/grgfil.f pgplot.new/src/grgfil.f
--- pgplot/src/grgfil.f	1995-06-14 03:23:31.000000000 +0200
+++ pgplot.new/src/grgfil.f	2008-11-13 13:01:36.000000000 +0100
@@ -10,13 +10,13 @@
 C     file specified by PGPLOT_FONT
 C     file "grfont.dat" in directory specified by PGPLOT_DIR
 C                       (with or without '/' appended)
-C     file "grfont.dat" in directory /usr/local/pgplot/
+C     file "grfont.dat" in directory /usr/share/pgplot/
 C
 C 2. The color-name database: try the following in order:
 C     file specified by PGPLOT_RGB
 C     file "rgb.txt" in directory specified by PGPLOT_DIR
 C                       (with or without '/' appended)
-C     file "rgb.txt" in directory /usr/local/pgplot/
+C     file "rgb.txt" in directory /usr/share/pgplot/
 C
 C Arguments:
 C  TYPE (input)  : either 'FONT' or 'RGB' to request the corresponding
@@ -26,7 +26,7 @@
 C  2-Dec-1994 - new routine [TJP].
 C-----------------------------------------------------------------------
       CHARACTER*(*) DEFDIR, DEFFNT, DEFRGB
-      PARAMETER  (DEFDIR='/usr/local/pgplot/')
+      PARAMETER  (DEFDIR='/usr/share/pgplot/')
       PARAMETER  (DEFFNT='grfont.dat')
       PARAMETER  (DEFRGB='rgb.txt')
       CHARACTER*255 FF
